TestModels/dafny-dependencies/StandardLibrary/runtimes/python/src/smithy_dafny_standard_library/internaldafny/extern/streams.py

```diff
@@ -0,0 +1,58 @@
+
+from _dafny import Seq
+from smithy_python.interfaces.blobs import ByteStream
+from smithy_dafny_standard_library.internaldafny.generated.Std_Streams import ByteStream as DafnyByteStream, RewindableByteStream as DafnyRewindableByteStream
+from smithy_dafny_standard_library.internaldafny.generated.Std_Enumerators import Enumerator
+from smithy_dafny_standard_library.internaldafny.generated.Std_Wrappers import Option, Option_Some, Option_None
+
+
+class DafnyByteStreamAsByteStream(ByteStream):
+  
+  def __init__(self, dafny_byte_stream):
+    self.dafny_byte_stream = dafny_byte_stream
+
+  def read(self, size: int = -1) -> bytes:
+    # TODO: assert size is -1, buffer, 
+    # or define a more specialized Action<int, bytes> type for streams.
+    next = self.dafny_byte_stream.Next()
+    while next.is_Some and len(next.value) == 0:
+      next = self.dafny_byte_stream.Next()
+    # Do NOT return None, because that indicates "no data right now, might be more later"
+    return bytes(next.value) if next.is_Some else bytes()
+
+
+class RewindableDafnyByteStreamAsByteStream(DafnyByteStreamAsByteStream):
+
+  def __init__(self, dafny_byte_stream):
+    if not isinstance(dafny_byte_stream, DafnyRewindableByteStream):
+      raise ValueError("Rewindable stream required")
+    super().__init__(dafny_byte_stream)
+
+  def tell(self) -> int:
+    return self.dafny_byte_stream.Position()
+
+  def seek(self, offset, whence=0):
+    match whence:
+      case 0:
+        position = offset
+      case 1:
+        position = self.dafny_byte_stream.Position() + offset
+      case 2:
+        position = self.dafny_byte_stream.ContentLength() + offset
+    return self.dafny_byte_stream.Seek(position)
+
+
+class StreamingBlobAsDafnyDataStream(DafnyByteStream):
```

Comment:
Just for others' clarification maybe:

```suggestion
class StreamingBlobAsDafnyDataStream(DafnyByteStream):
  """Wrapper class allowing a native streaming blob to be accessed from Dafny streaming code."""
```

Reply:
Fair, I'll add some more comments in general in fact